### PR TITLE
Add Brex to ADOPTERS.md

### PR DESCRIPTION
**Pull request by jrusso1020**
## Hey, I just made a Pull Request!

We have started using Backstage over the past quarter at Brex and
would like to add ourselves to the ADOPTERS.md!

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by jhaals** [comment]
Fantastic! Welcome @jrusso1020 and Brex 🥳

File: yarn.lock [hunk 1]
```diff
@@ -4236,7 +4236,7 @@
   dependencies:
     "@octokit/openapi-types" "^5.3.2"
 
-"@octokit/types@^6.13.0":
+"@octokit/types@^6.13.0", "@octokit/types@^6.8.2":
```
Comment (jhaals) [inline yarn.lock:4239]: Would it be possible to do this in another PR if it's needed at all? :)

Comment (benjdlambert) [reply yarn.lock:4239]: I think this is just a yarn lock drift. It's fixed in another PR. So gona merge :)

**Review by benjdlambert — APPROVED** [review]

File: yarn.lock (hunk 1 above)
Comment (jrusso1020) [reply yarn.lock:4239]: yeah sorry for the delay, this just happened when I ran `yarn install`
